doc/python_api_reference_vDev.md

```diff
@@ -2437,6 +2438,77 @@ def search_for_undetectable_logical_errors(
     """
 ```
 
+<a name="stim.Circuit.shortest_error_problem_as_wcnf_file"></a>
+```python
+# stim.Circuit.shortest_error_problem_as_wcnf_file
+
+# (in class stim.Circuit)
+def shortest_error_problem_as_wcnf_file(
+    self,
+    *,
+    weighted: bool = False,
+    weight_scale_factor: int = 0,
+) -> str:
+    """Generates a maxSAT problem instance in WDIMACS format whose optimal value is
+    the distance of the protocol, i.e. the minimum weight of any set of errors
+    that forms an undetectable logical error.
+    FYI: here is how you could fetch a solver from the
+    [2023 maxSAT competition](https://maxsat-evaluations.github.io/2023/) and
+    run the solver on a file produced with this method:
+    ```
+    # first download nzip CASHWMaxSAT-CorePlus.zip from 
+    # https://maxsat-evaluations.github.io/2023
+    unzip CASHWMaxSAT-CorePlus.zip
+    time ./CASHWMaxSAT-CorePlus/bin/cashwmaxsatcoreplus -bm -m problem.wcnf 
+    ```
+
+    Args:
+        weighted: Defaults to False (unweighted), so that the problem is to find
+        the minimum size set of errors that leads to an undetectable logical
+        error.
+        weight_scale_factor: Defaults to 0. If weighted==False, must set
+        weight_scale_factor to 0. If weighted==True, weight_scale_factor must be
+        set to a value >= 1. For a weighted problem, the errors will be quantized
+        into integers using this scale factor and the sum of the weights
+        will be minimized rather than the cardinality of the set of errors. For
+        a reasonably large quantization (weight_scale_factor > 100 is often
+        sufficient), the .wcnf file solution should be the (approximately)
+        most likely undetectable logical error. Note, however, that maxSAT solvers
+        can become slower when many distinct weights are provided, so for computing
+        the distance it is better to simply use the defaults weighted=False and
+        weight_scale_factor = 0.
+
+    Returns:
+        A WCNF file in [WDIMACS format](http://www.maxhs.org/docs/wdimacs.html)
+
+    Examples:
+        >>> import stim
+        >>> circuit = stim.Circuit("""
+        ...   X_ERROR(0.1) 0
+        ...   M 0
+        ...   OBSERVABLE_INCLUDE(0) rec[-1]
+        ...   X_ERROR(0.4) 0
+        ...   M 0
+        ...   DETECTOR rec[-1] rec[-2]
+        ... """)
+        >>> print(circuit.shortest_error_problem_as_wcnf_file(), end='')
+        p wcnf 2 4 5
+        1 -1 0
+        1 -2 0
+        5 -1 0
+        5 2 0
+        >>> print(circuit.shortest_error_problem_as_wcnf_file(
+        ...   weighted=True,
+        ...   weight_scale_factor=1000
+        ... ), end='')
+        p wcnf 2 4 4001
+        185 -1 0
+        1000 -2 0
+        4001 -1 0
+        4001 2 0
+    """
+```
+
 <a name="stim.Circuit.shortest_graphlike_error"></a>
 ```python
```

Comment:
This method should also be on `DetectorErrorModel`

Reply:
That sounds nice, but could we leave that for a future PR after we've solidified the API here?
Also, note that the heuristic approach `search_for_undetectable_logical_errors` is not a method of `DetectorErrorModel`. It seems like if we add the SAT method to `DetectorErrorModel` then why not also add the heuristic method.
WDYT?